controller/replica_controller.go

```diff
@@ -371,24 +371,10 @@ func (rc *ReplicaController) CreatePodSpec(obj interface{}) (*v1.Pod, error) {
 	// will pin it down to the same host because we have data on it
 	if r.Spec.NodeID != "" {
 		pod.Spec.NodeName = r.Spec.NodeID
-	} else {
-		pod.Spec.Affinity = &v1.Affinity{
-			PodAntiAffinity: &v1.PodAntiAffinity{
-				PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
-					{
-						Weight: 100,
-						PodAffinityTerm: v1.PodAffinityTerm{
-							LabelSelector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{
-									longhornReplicaKey: r.Spec.VolumeName,
-								},
-							},
-							TopologyKey: "kubernetes.io/hostname",
-						},
-					},
-				},
-			},
-		}
+	}
+	// error out if NodeID and DataPath wasn't filled in scheduler
+	if r.Spec.NodeID == "" || r.Spec.DataPath == "" {
+		return nil, fmt.Errorf("There has no avaible node for replica %v", r)
```

Comment:
Wrong error message. It's a bug.

`fmt.Errorf("BUG: Node or datapath wasn't set for replica %v", r.Name)`

Reply:
Pull this functional change into another commit or previous commit. It's kind of been smuggled into a commit suppose to add tests.